api/app/Controllers/Http/ChunkedUploadController.js

```diff
@@ -0,0 +1,99 @@
+"use strict";
+const Env = use("Env");
+const path = require("path");
+const fs = require("fs");
+const os = require("os");
+
+const TIMEOUT = 10*60000; // Timeout in milliseconds (30 seconds)
+
+class ChunkedUploadController {
+  deletePartialStoredChunks(totalChunks) {
+    const uploadPath = os.tmpdir();
+    for (let i = 0; i < totalChunks; i++) {
+      const tmpFilePath = path.join(uploadPath, `tempfile${index}.bin`);
```

Comment:
The expression `tempfile${index}.bin` is not random enough. Remember concurrent users uploading files will interfere here with the naming. Instead you may link the name to the current session, so you should find a way to link the filename against the session ID which can be derived from HTTP header, see latest changes of removing authentication layer, there is existing logic for that.

Reply:
Also the logic is not robust enough, you are simply counting to totalChunks instead of scanning the temporary directory for any chunk that exists and matches the following condition:
timeout exceeded of highest chunk order (means comparing time of last received chunk of a particular session, see file naming above)